src/cryptonote_core/service_node_rules.h

```diff
@@ -2,29 +2,30 @@
 
 #include "crypto/crypto.h"
 #include "cryptonote_config.h"
-#include "service_node_voting.h"
-
-#include <random>
+#include "cryptonote_core/service_node_voting.h"
 
 namespace service_nodes {
-  constexpr size_t   DEREGISTER_QUORUM_SIZE                    = 10;
+  constexpr size_t   UPTIME_QUORUM_SIZE                     = 10;
+  constexpr size_t   UPTIME_FRACTION_OF_THE_NETWORK_TO_TEST = 100;
+  constexpr size_t   UPTIME_MIN_NODES_TO_TEST               = 50;
+
   constexpr size_t   DEREGISTER_MIN_VOTES_TO_KICK_SERVICE_NODE = 7;
-  constexpr size_t   DEREGISTER_NTH_OF_THE_NETWORK_TO_TEST     = 100;
-  constexpr size_t   DEREGISTER_MIN_NODES_TO_TEST              = 50;
-  constexpr uint64_t DEREGISTER_VOTE_LIFETIME                  = BLOCKS_EXPECTED_IN_HOURS(2);
+  constexpr uint64_t VOTE_LIFETIME                             = 60;
 
   constexpr uint64_t CHECKPOINT_INTERVAL                       = 4;  // Checkpoint every 4 blocks and prune when too old except if (height % CHECKPOINT_STORE_PERSISTENTLY_INTERVAL == 0)
-  constexpr uint64_t CHECKPOINT_STORE_PERSISTENTLY_INTERVAL    = 60; // Persistently store the checkpoints at these intervals
-  constexpr uint64_t CHECKPOINT_VOTE_LIFETIME                  = CHECKPOINT_STORE_PERSISTENTLY_INTERVAL; // Keep the last 60 blocks worth of votes
+  constexpr uint64_t CHECKPOINT_STORE_PERSISTENTLY_INTERVAL    = VOTE_LIFETIME; // Persistently store the checkpoints at these intervals
+
 #if defined(LOKI_ENABLE_INTEGRATION_TEST_HOOKS)
-  constexpr size_t   CHECKPOINT_QUORUM_SIZE                    = 1;
-  constexpr size_t   CHECKPOINT_MIN_VOTES                      = 1;
+  constexpr ptrdiff_t MIN_TIME_IN_S_BEFORE_VOTING = 1;
+  constexpr size_t    CHECKPOINT_QUORUM_SIZE      = 1;
+  constexpr size_t    CHECKPOINT_MIN_VOTES        = 1;
 #else
-  constexpr size_t   CHECKPOINT_QUORUM_SIZE                    = 20;
-  constexpr size_t   CHECKPOINT_MIN_VOTES                      = 18;
+  constexpr ptrdiff_t MIN_TIME_IN_S_BEFORE_VOTING = 60 * 60 * 2;
+  constexpr size_t    CHECKPOINT_QUORUM_SIZE      = 20;
+  constexpr size_t    CHECKPOINT_MIN_VOTES        = 18;
```

Comment:
We should really use a variable somewhere for quorum supermajority and use that.  Some like:
```C++

constexpr int SUPER_MAJORITY_NUMERATOR = 2;
constexpr int SUPER_MAJORITY_DENOM = 3;
constexpr int supermajority_size(int total, int n = SUPER_MAJORITY_NUMERATOR, int d = SUPER_MAJORITY_DENOM) {
    return total * n / d + (total * n % d ? 1 : 0);
}
```

Reply:
I disagree, pre-empting to make it generic when its expressed fine as 1 value.